lib/oli_web/plugs/registration_captcha.ex

```diff
@@ -59,39 +59,55 @@ defmodule Oli.Plugs.RegistrationCaptcha do
     end
   end
 
-  defp render_captcha_error(conn, purpose) do
+  defp render_captcha_error(conn, :register_author) do
     conn =
       conn
       |> OliWeb.Pow.PowHelpers.use_pow_config(:author)
 
     changeset = Pow.Plug.change_user(conn, conn.params["user"])
     changeset = Ecto.Changeset.add_error(changeset, :captcha, "failed, please try again")
 
-    case purpose do
-      :register_author ->
-        conn
-        |> OliWeb.DeliveryController.render_create_form(
-          changeset: %{changeset | action: :insert},
-          sign_in_path: Routes.authoring_pow_session_path(conn, :new),
-          cancel_path: Routes.static_page_path(conn, :index)
-        )
-        |> halt()
-
-      :register ->
-        conn
-        |> OliWeb.DeliveryController.render_create_and_link_form(
-          changeset: %{changeset | action: :insert},
-          sign_in_path: Routes.authoring_pow_session_path(conn, :new),
-          cancel_path: Routes.static_page_path(conn, :index)
-        )
-        |> halt()
-
-      :create_and_link_account ->
-        conn
-        |> OliWeb.DeliveryController.render_create_and_link_form(
-          changeset: %{changeset | action: :insert}
-        )
-        |> halt()
-    end
+    conn
+    |> OliWeb.DeliveryController.render_create_form(
+      changeset: %{changeset | action: :insert},
+      sign_in_path: Routes.authoring_pow_session_path(conn, :new),
+      cancel_path: Routes.static_page_path(conn, :index)
+    )
+    |> halt()
+  end
+
+  defp render_captcha_error(conn, :register) do
+    conn =
+      conn
+      |> OliWeb.Pow.PowHelpers.use_pow_config(:user)
+
+    {:error, changeset} =
+      Pow.Plug.change_user(conn, conn.params["user"])
+      |> Ecto.Changeset.add_error(:captcha, "failed, please try again")
+      |> Ecto.Changeset.apply_action(:insert)
+
+    conn
+    |> assign(:changeset, changeset)
+    |> assign(:action, Routes.pow_registration_path(conn, :create))
+    |> assign(:sign_in_path, Routes.pow_session_path(conn, :new))
+    |> assign(:cancel_path, Routes.delivery_path(conn, :index))
+    |> Phoenix.Controller.put_view(OliWeb.Pow.RegistrationView)
+    |> Phoenix.Controller.render("new.html")
+    |> halt()
```

Comment:
I don't like this bit.

The other 2 cases had a function inside DeliveryController to call, but I didn't see an equivalent place for a non-authoring account. I could go create a new controller with just this function in it if people prefer, but that seemed weird too.

Open to suggestions.

Reply:
I think that DeliveryController is a good place where to define a `render_user_create_form` function since define some other "user functions". With that, we should also change the above's name to include `author` or something like that

Reply:
That's done now, thanks!